client/planning-extension/src/authoring-react-fields/coverages/editor.tsx

```diff
@@ -5,101 +5,100 @@ import {
     IUrlsFieldUserPreferences,
 } from 'superdesk-api';
 import {ICoveragesValueOperational} from './interfaces';
-import {cloneDeep, set} from 'lodash';
 import {extensionBridge} from '../../extension_bridge';
 import {IPlanningItem} from '../../../../interfaces';
 import {superdesk} from '../../superdesk';
+import {DebouncedChangeHOC} from '../debounced-change-hoc';
+import {cloneDeep, set} from 'lodash';
 
 type IProps = IEditorComponentProps<ICoveragesValueOperational, IUrlsFieldConfig, IUrlsFieldUserPreferences>;
 
 export class Editor extends React.PureComponent<IProps> {
     render() {
         const Container = this.props.container;
-
         const {EditorFieldCoverages} = extensionBridge.editor.fields;
+        const itemSaved = (this.props.item._id != null) && (this.props.item._id ?? '').startsWith('temp-') === false;
 
         return (
-            <EditorFieldCoverages
-                field="coverages"
-
-                item={{
-                    // coverages are the main value
-                    coverages: this.props.value,
-
-                    // related_events are used if available to prefill coverage fields when adding a new coverage
-                    related_events: this.props.item.related_events,
-                } as IPlanningItem}
+            <DebouncedChangeHOC
 
                 /**
-                 * It looks like this prop is designed to accept a validation message.
-                 * authoring-react field types don't accept validation messages.
-                 * They are rendered higher in the component tree.
-                 * We do handle it in PlanningEditorStandalone component (on save).
+                 * Update component when value from props changes. Otherwise the component won't
```

Comment:
With the comment now it's even more clear that this should be handled inside the component.

When returning components from render method we have an assumption that components will respect props - and that's how it should be. If some components don't respect props - we need to fix it inside the components itself. We have situations when it's best to simply remount it, but let's do a remounting inside a component itself via a wrapper component. Similar how we connect redux by creating a wrapper component.